### ISSUE TITLE
find_proto function should cache results

### ISSUE THREAD

**Issue by rossengeorgiev** [ValvePython/dota2#11]
https://github.com/ValvePython/dota2/blob/eb8d814b6404448a0fe40f2ebb37e020cac0edda/dota2/msg.py#L46

**Comment by DeV1doR** [comment]
@rossengeorgiev, how about default functools cache https://gist.github.com/DeV1doR/366d7135559763dade0cd4940f14ba48 ?

**Comment by rossengeorgiev** [comment]
Yep, that's one option. Not sure if I want to add an unmaintained backport. `cachetools` seems like a much better choice.
